### PR TITLE
Df.append() add index handling

### PR DESCRIPTION
**Pull request by akharche**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -261,11 +263,7 @@ def sdc_pandas_dataframe_append_impl(df, other, ignore_index=True, verify_integr
     func_args = ['df', 'other']
 
     for key, value in args:
-        # TODO: improve check
-        if key not in func_args:
-            if isinstance(value, types.Literal):
-                value = value.literal_value
-            func_args.append(f'{key}={value}')
+        func_args.append(f'{key}={value}')
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: That is equal to:
```python
func_args = ['df', 'other'] + kwsparams2list(args)
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -316,27 +313,41 @@ def sdc_pandas_dataframe_append_impl(df, other, ignore_index=True, verify_integr
...
-    # TODO: Handle index
-    func_text.append(f"return pandas.DataFrame({{{data}}})\n")
+
+    if ignore_index_value == True:  # noqa
+        func_text.append(f'return pandas.DataFrame({{{data}}})\n')
+    else:
+        func_text.append(f'df_index = df.index')
+        func_text.append(f'other_index = other.index')
+        func_text.append(f'joined_index = hpat_arrays_append(df_index, other_index)\n')
+        func_text.append(f'return pandas.DataFrame({{{data}}}, index=joined_index)\n')
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: You could decrease number of words:
```suggestion
        func_text += [f'df_index = df.index',
                      f'other_index = other.index',
                      f'joined_index = hpat_arrays_append(df_index, other_index)\n',
                      f'return pandas.DataFrame({{{data}}}, index=joined_index)\n']
```
or
```suggestion
        func_text += [f'joined_index = hpat_arrays_append(df.index, other.index)\n',
                      f'return pandas.DataFrame({{{data}}}, index=joined_index)\n']
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 3]
```diff
@@ -387,27 +398,43 @@ def sdc_pandas_dataframe_append(df, other, ignore_index=True, verify_integrity=F
...
         ty_checker.raise_exc(sort, 'boolean, None', 'sort')
 
-    args = (('ignore_index', True), ('verify_integrity', False), ('sort', None))
+    if not isinstance(ignore_index, (bool, types.Boolean, types.Omitted)):
+        ty_checker.raise_exc(ignore_index, 'boolean', 'ignore_index')
+
+    none_or_numeric_indexes = ((isinstance(df.index, types.NoneType) or isinstance(df.index, types.Number))
+                               and (isinstance(other.index, types.NoneType) or isinstance(other.index, types.Number)))
+
+    indexes_comparable = check_types_comparable(df.index, other.index) or none_or_numeric_indexes
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: You could move these lines to elif-block and use it only there if the condition is True.

**Comment by AlexanderKalistratov** [comment]
@akharche  @densmirn  Does this PR affects documentation?

**Comment by akharche** [comment]
@AlexanderKalistratov Yes, I'm waiting for merging Maria's PR that fixes append documentation and then I do rebase with improving Limitation block in docs

**Pull request merged**